libbeat/common/docker/watcher.go

```diff
@@ -290,7 +291,20 @@ func (w *watcher) listContainers(options types.ContainerListOptions) ([]*Contain
 	for _, c := range containers {
 		var ipaddresses []string
 		for _, net := range c.NetworkSettings.Networks {
-			ipaddresses = append(ipaddresses, net.IPAddress)
+			if net.IPAddress != "" {
+				ipaddresses = append(ipaddresses, net.IPAddress)
+			}
+		}
+
+		// If there are no network interfaces, assume that the container is on host network
+		// Inspect the container directly and use the hostname as the IP address in order
+		if len(ipaddresses) == 0 {
+			info, err := w.client.ContainerInspect(w.ctx, c.ID)
+			if err == nil {
+				ipaddresses = append(ipaddresses, info.Config.Hostname)
+			} else {
+				logp.Debug("docker", "unable to inspect container %s due to error %v", c.ID, err)
```

Comment:
this should probably be a warning message? as it should not happen frequently, and can cause some debugging nightmares